rest/route/user.go

```diff
@@ -405,18 +408,31 @@ func (h *userRolesPostHandler) Run(ctx context.Context) gimlet.Responder {
 	}
 	dbUser, valid := u.(*user.DBUser)
 	if dbUser == nil {
-		return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
-			Message:    fmt.Sprintf("no matching user for '%s'", h.userID),
-			StatusCode: http.StatusNotFound,
-		})
+		if h.CreateUser {
+			um := evergreen.GetEnvironment().UserManager()
+			newUser := user.DBUser{
+				Id:          h.userID,
+				SystemRoles: h.Roles,
+			}
+			u, err = um.GetOrCreateUser(&newUser)
+			if err != nil {
+				return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{StatusCode: http.StatusInternalServerError, Message: fmt.Sprintf("unable to create new user: %s'", err.Error())})
+			}
+			return gimlet.NewJSONResponse(struct{}{})
+		} else {
+			return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+				Message:    fmt.Sprintf("no matching user for '%s'", h.userID),
+				StatusCode: http.StatusNotFound,
+			})
+		}
 	}
 	if !valid {
```

Comment:
This isn't in the PR, but since I noticed, can we just check `valid` or remove it entirely since it doesn't do anything to check `valid` after checking for `dbUser == nil`?

Reply:
moved it up